### PR TITLE
Fix react newline issues

### PR DESCRIPTION
**Pull request by stevemk14ebr**
Fixes https://github.com/rehypejs/rehype-react/issues/28

### CODE REVIEW

**Comment by wooorm** [comment]
Thanks!

Looks like the formatting is off. Could you `npm install` and `npm test` locally?

File: index.js [hunk 1]
```diff
@@ -35,6 +36,23 @@ function rehypeReact(options) {
...
     var component = name
 
+    // Currently, a warning is triggered by react for *any* white space in
+    // tables.
+    // So we remove the pretty lines for now.
+    // See: <https://github.com/facebook/react/pull/7081>.
+    // See: <https://github.com/facebook/react/pull/7515>.
+    // See: <https://github.com/remarkjs/remark-react/issues/64>.
+    if (children && tableElements.indexOf(name) !== -1) {
+        children = children.filter(function (child) {
```
Comment (wooorm) [inline index.js:46]: I think `npm test` will also format this code, could you do that?

File: index.js [hunk 2]
```diff
@@ -35,6 +36,23 @@ function rehypeReact(options) {
...
+            const notNewline = child !== '\n' && child !== '\r';
+            let notEmpty = true;
+            if (typeof child === 'string') {
+                notEmpty = child.trim().length > 0;
+            }
+            return notNewline && notEmpty;
```
Comment (wooorm) [inline index.js:52]: Can you instead use the functionality that is used here: https://github.com/remarkjs/remark-react/blob/d4311dd0f91813b08d0ca232f112b420e1ac0979/index.js#L47-L49. Removing *just* `'\n'`?

Comment (stevemk14ebr) [reply index.js:52]: why? I've seen tables created that have space nodes that look like they are also for formatting and do not relate to the content put into the table

Comment (stevemk14ebr) [reply index.js:52]: ex: https://github.com/rehypejs/rehype-react/issues/28#issuecomment-862388914

Comment (wooorm) [reply index.js:52]: I only see `\n` in those cases (and applying this fix to several projects has not been a problem). Could you make that reproducible?

Comment (stevemk14ebr) [reply index.js:52]: apologies it is just \n, the codesandbox didn't render it escaped so it was "\n" in reality but looked empty. I will adjust this

File: index.js (hunk 1 above)
Comment (wooorm) [reply index.js:46]: @stevemk14ebr Could you run `npm test` on your code to format it matching the project’s style?

Comment (stevemk14ebr) [reply index.js]: done!

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Released! https://github.com/rehypejs/rehype-react/releases/tag/6.2.1

**Pull request merged**